src/dos/dos_classes.cpp

```diff
@@ -308,6 +309,119 @@ bool DOS_PSP::SetNumFiles(uint16_t file_num)
 	return true;
 }
 
+static constexpr auto bytes_to_read = 1024;
+
+std::optional<std::string> DOS_PSP::GetEnvStr(const std::string_view entry) const
+{
+	/* Walk through the internal environment and see for a match */
+	PhysPt env_read = PhysicalMake(GetEnvironment(), 0);
+
+	char env_string[bytes_to_read + 1];
+	if (entry.empty()) {
+		return {};
+	}
+
+	for (;;) {
+		MEM_StrCopy(env_read, env_string, bytes_to_read);
+		if (!env_string[0]) {
+			return {};
+		}
+		env_read += (PhysPt)(safe_strlen(env_string) + 1);
+		char* equal = strchr(env_string, '=');
+		if (!equal) {
+			continue;
+		}
+		/* replace the = with \0 to get the length */
+		*equal = '\0';
+		if (strlen(env_string) != entry.size()) {
+			continue;
+		}
+		if (!iequals(entry, env_string)) {
+			continue;
+		}
+
+		return env_string + entry.size() + sizeof('=');
+	}
+}
+
+std::vector<std::string> DOS_PSP::GetAllEnvVars() const
```

Comment:
Does this list the names of all env vars? So just the key names (variable names), but not the values, right? Could we use unambiguous and consistent naming across the board? By reading these method names and the names of their arguments, and having no familiarity with this code, I'm a bit confused.

```
std::optional<std::string> DOS_PSP::GetEnvStr(const std::string_view entry) const
```
So `entry` is the name of the variable? Can we use `var_name` instead of `entry` then?

I guess my point is, we should pick one single name for these things and then stick with it everywhere, method names, arg names, variable names in the code, etc. 

I know some of this is pre-existing, I guess I'm just saying let's try to get it right now if you went to the trouble of touching all this code 😄